src/Chapter3.hs

```diff
@@ -480,6 +514,49 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
 
+data City = City
+  { cityCastle       :: Castle
+  , cityMainBuilding :: Building
+  , cityHouses       :: [Building]
+  }
+
+data Castle
+  = WithoutCastle
+  | JustCastle
+  | SecureCastle
+
+data Building
+  = Church
+  | Library
+  | BuildingForOne
+  | BuildingForTwo
+  | BuildingForThree
+  | BuildingForFour
```

Comment:
This is a nice sum type that represents the whole kind of buildings!

However, I see a small problem with such approach:
 * `cityMainBuilding` could be `BuildingForOne` (but it should be either Church or Library, and there are no guarantees that it would be just that)
 * `cityHouses` can contain main buildings

This could be resolved if you separate this into two different sum types: `MainBuilding` and `Building` 🏗️